app/routes/users+/$username_+/articles.tsx

```diff
@@ -0,0 +1,100 @@
+import { json, type LoaderFunctionArgs } from '@remix-run/node'
+import { Link, NavLink, Outlet, useLoaderData } from '@remix-run/react'
+import { GeneralErrorBoundary } from '~/components/ErrorBoundary.js'
+import { Icon } from '~/components/ui/icon.tsx'
+import { prisma } from '~/utils/db.server.ts'
+import { cn, getUserImgSrc } from '~/utils/misc.tsx'
+import { useOptionalUser } from '~/utils/user.ts'
+
+export async function loader({ params }: LoaderFunctionArgs) {
+	const owner = await prisma.user.findFirst({
+		select: {
+			id: true,
+			name: true,
+			username: true,
+			image: { select: { id: true } },
+			articles: { select: { id: true, title: true } },
+		},
+		where: { username: params.username },
+	})
+
+	invariantResponse(owner, 'Owner not found', { status: 404 })
+
+	return json({ owner })
+}
+
+export default function ArticlesRoute() {
+	const data = useLoaderData<typeof loader>()
+	const user = useOptionalUser()
+	const isOwner = user?.id === data.owner.id
+	const ownerDisplayName = data.owner.name ?? data.owner.username
+	const navLinkDefaultClassName =
+		'line-clamp-2 block rounded-l-full py-2 pl-8 pr-6 text-base lg:text-xl'
+	return (
+		<main className="container flex h-full min-h-[400px] px-0 pb-12 md:px-8">
+			<div className="grid w-full grid-cols-4 bg-muted pl-2 md:container md:rounded-3xl md:pr-0">
+				<div className="relative col-span-1">
+					<div className="absolute inset-0 flex flex-col">
+						<Link
+							to={`/users/${data.owner.username}`}
+							className="flex flex-col items-center justify-center gap-2 bg-muted pb-4 pl-8 pr-4 pt-12 lg:flex-row lg:justify-start lg:gap-4"
+						>
+							<img
+								src={getUserImgSrc(data.owner.image?.id)}
+								alt={ownerDisplayName}
+								className="h-16 w-16 rounded-full object-cover lg:h-24 lg:w-24"
+							/>
+							<h1 className="text-center text-base font-bold md:text-lg lg:text-left lg:text-2xl">
+								{ownerDisplayName}'s Articles
+							</h1>
+						</Link>
+						<ul className="overflow-y-auto overflow-x-hidden pb-12">
+							{isOwner ? (
+								<li className="p-1 pr-0">
+									<NavLink
+										to="new"
+										className={({ isActive }) =>
+											cn(navLinkDefaultClassName, isActive && 'bg-accent')
+										}
+									>
+										<Icon name="plus">New Article</Icon>
+									</NavLink>
+								</li>
+							) : null}
+							{data.owner.articles.map(article => (
+								<li key={article.id} className="p-1 pr-0">
+									<NavLink
+										to={article.id}
+										preventScrollReset
+										prefetch="intent"
+										className={({ isActive }) =>
+											cn(navLinkDefaultClassName, isActive && 'bg-accent')
+										}
+									>
+										{article.title}
+									</NavLink>
+								</li>
+							))}
+						</ul>
+					</div>
+				</div>
+				<div className="relative col-span-3 bg-accent md:rounded-r-3xl">
+					<Outlet />
+				</div>
+			</div>
+		</main>
+	)
+}
```

Comment:
This works well, but you've forgotten to add a link to the user's articles from their profile page. See [this step](https://gedsargent.github.io/code-monster/guides/epic-news/displaying-articles/creating-articles/#add-an-articles-page-link) on Code Monster for help.